src/utilities/valueToLiteral.js

```diff
@@ -0,0 +1,97 @@
+import { inspect } from '../jsutils/inspect';
+import { isIterableObject } from '../jsutils/isIterableObject';
+
+import type { ConstValueNode } from '../language/ast';
+import { Kind } from '../language/kinds';
+
+import type { GraphQLInputType } from '../type/definition';
+import {
+  getNamedType,
+  isLeafType,
+  isInputObjectType,
+} from '../type/definition';
+
+/**
+ * Produces a GraphQL Value AST given a JavaScript value.
+ *
+ * A GraphQL type may be provided, which will be used to interpret different
+ * JavaScript values if it defines a `valueToLiteral` method.
+ *
+ * | JavaScript Value  | GraphQL Value        |
+ * | ----------------- | -------------------- |
+ * | Object            | Input Object         |
+ * | Array             | List                 |
+ * | Boolean           | Boolean              |
+ * | String            | String               |
+ * | Number            | Int / Float          |
+ * | null / undefined  | Null                 |
+ *
+ * Note: This function does not perform any type validation or coercion.
+ */
+export function valueToLiteral(
+  value: mixed,
+  type?: GraphQLInputType,
+): ConstValueNode {
+  // Like JSON, a null literal is produced for null and undefined.
+  if (value == null) {
+    return { kind: Kind.NULL };
+  }
+
+  const namedType = type && getNamedType(type);
+
+  // Convert JavaScript array to GraphQL list.
+  if (isIterableObject(value)) {
+    return {
+      kind: Kind.LIST,
+      values: Array.from(value, (item) => valueToLiteral(item, namedType)),
+    };
+  }
+
+  // Does this type (if provided) define `valueToLiteral` which returns a value?
```

Comment:
what about input object types? Does InputObject define a `valueToLiteral`? It either should or we should define the appropriate logic here.

Reply:
No, only leaf types, similar to coercion. Input Objects should never have overridden behavior they should always use the behavior described here.